types.go

```diff
@@ -220,6 +235,22 @@ func (p *Permission) Validate(vr *ValidationResults) {
 	}
 }
 
+// SubscribePermission defines allow/deny subjects used for subscriptions.
+type SubscribePermission struct {
+	Allow StringList `json:"allow,omitempty"`
+	Deny  StringList `json:"deny,omitempty"`
+}
+
+// Validate the allow, deny elements of a permission
+func (p *SubscribePermission) Validate(vr *ValidationResults) {
```

Comment:
the validation doesn't really work because if the subscribe permission is `>.a queue` the invalid subject is not validated. again this is the problem with this strategy, we cannot really validate, because if the user intended the subject `a b`, it would be interpreted as subject `a` queue `b`

Reply:
The data model for expressing this is not correct - it cannot just add a space when one of the components specifies a space as illegal.

Reply:
`>.a` is an invalid subject so we could add the `IsValidLiteralSubject` logic from the server here though would help the most in `nsc` to catch these earlier rather than during runtime which would only result in an authentication error rather than invalid creds for example.  I think the model is fine I'd say that also comparatively easier to maintain than making the involved types more complex by moving from subject string to object type with subject  + queue.